### PR TITLE
Fix flaky test MasterServiceTests.testClusterStateUpdateLoggingWithDebugEnabled

### PR DESCRIPTION
**Pull request by sumitasr**
### Description
Fix flaky test MasterServiceTests.testClusterStateUpdateLoggingWithDebugEnabled

+ Fixed assertions.

### Related Issues
Resolves #[15117](https://github.com/opensearch-project/OpenSearch/issues/15117)

### Check List
- [NA] Functionality includes testing.
- [NA] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [Yes] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/test/java/org/opensearch/cluster/service/MasterServiceTests.java [hunk 1]
```diff
@@ -698,9 +698,6 @@ public void onFailure(String source, Exception e) {
                     }
                 });
                 assertBusy(mockAppender::assertAllExpectationsMatched);
-                // verify stats values after state is published
-                assertEquals(1, clusterManagerService.getClusterStateStats().getUpdateSuccess());
-                assertEquals(0, clusterManagerService.getClusterStateStats().getUpdateFailed());
```
Comment (owaiskazi19) [inline server/src/test/java/org/opensearch/cluster/service/MasterServiceTests.java:703]: Removing tests shouldn't be a way to fix the failing flaky tests here

Comment (sumitasr) [reply server/src/test/java/org/opensearch/cluster/service/MasterServiceTests.java:703]: Agreed @owaiskazi19, Intention of this unit test was only to check the correct logging statements at the debug level in the submit path. It is already being done in ```assertBusy(mockAppender::assertAllExpectationsMatched);```

We don't want to assert the end publish state. So, removed it.

Comment (sumitasr) [reply server/src/test/java/org/opensearch/cluster/service/MasterServiceTests.java:703]: We could have done something like below to make it non flaky as well. But, i thought to remove it.

```
assertBusy(() -> assertEquals(1, clusterManagerService.getClusterStateStats().getUpdateSuccess()));
assertBusy(() -> assertEquals(0, clusterManagerService.getClusterStateStats().getUpdateFailed()));
```

**Comment by andrross** [comment]
@sumitasr I don't think we saw this test failing in the past. What changed? I would feel better about removing the assertions here if we fully understand what changed to make this test intermittently fail when in the past it never failed.

**Comment by sumitasr** [comment]
> @sumitasr I don't think we saw this test failing in the past. What changed? I would feel better about removing the assertions here if we fully understand what changed to make this test intermittently fail when in the past it never failed.

These 2 assertions were flaky in the past as well in a different test and engineer moved the assertions from different tests into logging verification hoping for stable test. And recently we created 2 tests to verify trace and debug logging behavior independently. This test started getting tagged as flaky. 

I ran assertions offline to verify before assertion both getUpdateSuccess and getUpdateFailed is reported as 0 during flaky behavior. The ClusterStateStats is updated in async path and can be delayed. Added assertsBusy to make sure we get consistent behavior in the test which was not added earlier. 

Ran the test 5000 times.

**Comment by sumitasr** [comment]
> ❌ Gradle check result for [98ffaa7](https://github.com/opensearch-project/OpenSearch/commit/98ffaa7cf8503596ad83fbbc576d12da1186ecbb): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/45684/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

Fixed

**Comment by sumitasr** [comment]
> ❕ Gradle check result for [a32ae3a](https://github.com/opensearch-project/OpenSearch/commit/a32ae3a7b230e2d317e8e99296ab3045fe18d43b): [UNSTABLE](https://build.ci.opensearch.org/job/gradle-check/45765/)
> 
> Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

Failed due to [Test Result](https://build.ci.opensearch.org/job/gradle-check/45765/testReport/) (2 failures / +2)

    [org.opensearch.search.SearchTimeoutIT.testSimpleTimeout {p0={"search.concurrent_segment_search.enabled":"false"}}](https://build.ci.opensearch.org/job/gradle-check/45765/testReport/junit/org.opensearch.search/SearchTimeoutIT/testSimpleTimeout__p0___search_concurrent_segment_search_enabled___false___/)
    [org.opensearch.search.SearchTimeoutIT.testSimpleTimeout {p0={"search.concurrent_segment_search.enabled":"true"}}](https://build.ci.opensearch.org/job/gradle-check/45765/testReport/junit/org.opensearch.search/SearchTimeoutIT/testSimpleTimeout__p0___search_concurrent_segment_search_enabled___true___/)

Need to retry build

**Review by shwetathareja — APPROVED** [review]

**Pull request merged**